scripts/import_data.js

```diff
@@ -0,0 +1,124 @@
+'use strict';
+
+const _ = require('lodash');
+const fs = require('fs');
+const readline = require('readline');
+const config = require('./../config');
+const tenderExtractor = require('./../extractors/tender');
+const buyerExtractor = require('./../extractors/buyer');
+
+// Alternative implementation where the update/create logic happens in the `then` of `one()`
+// Same problem as below, `existingRecord` is always undefined thus `update` never happens and the `created` done instead fails
+function upsertVertex(transaction, className, attrs) {
+  const recordName = `${className.toLowerCase()}${attrs.id.replace(/-/g, '')}`;
+  config.db.select().from(className).where({ id: attrs.id }).one()
+    .then((existingRecord) => {
+      transaction.let(recordName, (t) => {
+        if (_.isUndefined(existingRecord)) {
+          transaction.create('vertex', className)
+            .set(attrs);
+        } else {
+          t.update(className)
+            .set(attrs)
+            .where({ id: attrs.id })
+            .return('AFTER');
+        }
+      });
+    });
+}
+
+function writeTender(tenderAttrs) {
+  const tender = tenderExtractor.extractTender(tenderAttrs);
+  function recordName(uuid, className) {
+    return `${className.toLowerCase()}${uuid.replace(/-/g, '')}`;
+  }
+  const tenderName = recordName(tender.id, 'Tender');
+
+  // `existingTender` is always undefined thus `update` never happens and the `create` done instead fails
+  let existingTender;
+  config.db.select().from('Tender').where({ id: tender.id }).one()
+    .then((t) => { existingTender = t; });
```

Comment:
It's really frustrating that when I do this in the console it fetches the existing tender.

Reply:
I figured why this happens and solved it in the [next commit](https://github.com/tenders-exposed/elvis-backend-node/pull/21/commits/1d773e71511b4ea2a0dad9ec2e54f23d574c7e46) by moving the create/update logic in `then()`.